### PR TITLE
Fix docs for v0.2, add sndfile

### LINKED ISSUE #192: OSError: sndfile library not found during docs build
Spotted by @r4b1tt on latest master today. Unintended consequence of #183, which went from `librosa<0.7.0` to `librosa>=0.7.0`

Possibly related: https://github.com/librosa/librosa/issues/1017

Traceback below:

```python
mirdata.gtzan_genre
^^^^^^^^^^^^^^^
WARNING: autodoc: failed to import module 'gtzan_genre' from module 'mirdata'; the following exception was raised:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/mirdata/envs/latest/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 154, in import_module
    __import__(modname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/mirdata/checkouts/latest/mirdata/gtzan_genre.py", line 18, in <module>
    import librosa
  File "/home/docs/checkouts/readthedocs.org/user_builds/mirdata/envs/latest/lib/python3.7/site-packages/librosa/__init__.py", line 12, in <module>
    from . import core
  File "/home/docs/checkouts/readthedocs.org/user_builds/mirdata/envs/latest/lib/python3.7/site-packages/librosa/core/__init__.py", line 126, in <module>
    from .audio import *  # pylint: disable=wildcard-import
  File "/home/docs/checkouts/readthedocs.org/user_builds/mirdata/envs/latest/lib/python3.7/site-packages/librosa/core/audio.py", line 10, in <module>
    import soundfile as sf
  File "/home/docs/checkouts/readthedocs.org/user_builds/mirdata/envs/latest/lib/python3.7/site-packages/soundfile.py", line 142, in <module>
    raise OSError('sndfile library not found')
OSError: sndfile library not found
```

Proposed fix: explicitly add soundfile to docs/requirements.txt

### PR DESCRIPTION
**Pull request by lostanlen**
Fixes #192 

@rabitt can you do `pip install -e .[docs]` on a fresh environment and tell me if this works on your machine?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -50,10 +50,12 @@
                 'dali-dataset',
             ],
             'docs': [
+                'DALI-dataset>=1.0.0',
```
Comment (rabitt) [inline setup.py]: does dali-dataset need to also be here? it's already in docs/requirements.txt

Comment (lostanlen) [reply setup.py]: good question. i'm not sure. i tried the following:

`pip install .[docs]`
`cd docs`
`make html`

and it passed. it didn't pass without `DALI-dataset`. But perhaps i'm making docs the wrong way? How is `docs/requirements.txt` linked exactly?

Comment (rabitt) [reply setup.py]: I'm not sure either - it gets picked up by readthedocs. Let's merge as-is, it doesn't really hurt to have an extra requirement in the docs build.

**Comment by rabitt** [comment]
I think I found the problem - in `docs/requirements.txt`:
`soundfile` should be `sndfile`

**Pull request merged**